### PR TITLE
Implement Hashing to Reduce Database Bloat in Sync Command

### PR DESCRIPTION
**Pull request by sanjacornelius**
This PR implements hashing for the helper process and process template to reduce database bloat when running the sync command. The hashing mechanism generates a unique hash for each process, which is then used to check for any changes within the processes. If the hash has changed since the last sync, indicating that the process has been modified, only then are the processes imported into the instance. This approach prevents the processes from being imported every time the sync command is run, thereby eliminating unintended database bloat, particularly in the asset versions table.

### CODE REVIEW

**Review by estebangallego — APPROVED** [review]

File: .github/workflows/scripts/generate_index.php [hunk 1]
```diff
@@ -15,19 +15,16 @@ function update_readme($categories) {
         $category = ucwords($category);
         fwrite($readme, "\n## $category\n");
         // Sort templates alphabetically within each category
-        usort($templates, function($a, $b) { return strcmp($a['name'], $b['name']); });
+        usort($templates, function($a, $b) { return strcmp($a['template_details']['card-title'], $b['template_details']['card-title']); });
```
Comment (ryancooley) [inline .github/workflows/scripts/generate_index.php]: @sanjacornelius There are some pretty deeply nested array paths here. Should we add some checks to make sure the expected keys actually exist?

File: .github/workflows/scripts/generate_index.php [hunk 2]
```diff
@@ -187,9 +189,13 @@ function mapContentToTemplateStructure($contentInfo, &$categories, $currentCateg
 
     switch ($fileName) {
         case "process_helper_export":
+            $data = json_decode(file_get_contents($contentInfo->getPathname()), true);
+            $categories[$currentCategory][$templateName]['template_details']['helper_process_hash'] = compute_hash(json_encode($data['export'][$data['root']]['attributes']));
```
Comment (ryancooley) [inline .github/workflows/scripts/generate_index.php]: Same here.

File: .github/workflows/scripts/generate_index.php [hunk 3]
```diff
@@ -187,9 +189,13 @@ function mapContentToTemplateStructure($contentInfo, &$categories, $currentCateg
...
             $categories[$currentCategory][$templateName]['helper_process'] = $contentInfo->getPathname();
             break;
         case "process_template_export":
+            $data = json_decode(file_get_contents($contentInfo->getPathname()), true);
+            $categories[$currentCategory][$templateName]['template_details']['template_process_hash'] = compute_hash(json_encode($data['export'][$data['root']]['attributes']));
```
Comment (ryancooley) [inline .github/workflows/scripts/generate_index.php]: And here.

**Review by ryancooley — CHANGES_REQUESTED** [review]

File: .github/workflows/scripts/generate_index.php (hunk 1 above)
Comment (sanjacornelius) [reply .github/workflows/scripts/generate_index.php]: Hey @ryancooley sure I'll add some checks!

File: .github/workflows/scripts/generate_index.php (hunk 2 above)
Comment (sanjacornelius) [reply .github/workflows/scripts/generate_index.php]: 👍

File: .github/workflows/scripts/generate_index.php (hunk 3 above)
Comment (sanjacornelius) [reply .github/workflows/scripts/generate_index.php]: For the `$categories` array, there's a function called `initializeTemplateStructure()` which is used to assign values. 

Here's an example of how it's used: 
`$categories[$currentCategory][$templateName] = initializeTemplateStructure();`

This function initializes all the necessary keys in the `$categories` array. This ensures that the structure of the `$categories` is properly set up. 

Here's what the `initializeTemplateStructure()` function looks like:

```
function initializeTemplateStructure()
{
    return [
        "helper_process" => "",
        "template_process" => "",
        "config_collection" => "",
        "template_details" => [
            "card-title" => "",
            "card-excerpt" => "",
            "modal-excerpt" => "",
            "modal-description" => "",
            'version' => "",
            'unique-template-id' => "",
            "helper_process_hash" => "",
            "template_process_hash" => "",
        ],
        "assets" => [
            "icon" => "",
            "card-background" => "",
            "slides" => [],
            "launchpad"  => [
                "process-card-background" => "",
                "slides" => [],
            ]
        ],
        "connected_accounts" => []
    ];
}
```

**Comment by sanjacornelius** [comment]
Hey @ryancooley code has been updated. Can you please review?

**Pull request closed without merging**